src/Lecture3.hs

```diff
@@ -246,4 +330,26 @@ Just [8,9,10]
 [8,20,3]
 
 -}
-apply = error "TODO"
+apply :: Functor t => a -> t (a -> b) -> t b
+apply e f = fmap ($ e) f 
```

Comment:
You can't pattern match on a constructor of a polymorphic type because you don't know the constructor name. But if you didn't have `$`, you could use explicit lambda here:

```suggestion
apply e f = fmap (\f -> f e) f 
```

Reply:
wow! thank you